controllers/virtualmachinesnapshot/virtualmachinesnapshot_controller.go

```diff
@@ -319,6 +297,105 @@ func ensureCSIVolumeSyncAnnotation(vmSnapshot *vmopv1.VirtualMachineSnapshot) {
 	}
 	// As long as the value is not set to completed by CSI driver, we need to mark it as requested.
 	if vmSnapshot.Annotations[constants.CSIVSphereVolumeSyncAnnotationKey] != constants.CSIVSphereVolumeSyncAnnotationValueCompleted {
-		vmSnapshot.Annotations[constants.CSIVSphereVolumeSyncAnnotationKey] = constants.CSIVSphereVolumeSyncAnnotationValueRequest
+		vmSnapshot.Annotations[constants.CSIVSphereVolumeSyncAnnotationKey] = constants.CSIVSphereVolumeSyncAnnotationValueRequested
+	}
+}
+
+func setSnapshotReadyCondition(vmSnapshot *vmopv1.VirtualMachineSnapshot) {
```

Comment:
maybe call it reconcileSnapshotConditions since you are essentially manipulating three conditions here, Ready being one of them.